azavea/raster-foundry
# Processing Status

This PR incorporates the new processing block and wires it up. The app polls the backend to get layers that have a `status` field set to `uploading` or `processing` or that have been set to `complete` since the last time the frontend was refreshed. It is the responsibility of services running on the worker VM to update the `status` and `status_updated_at` fields. See #125.

##### Testing
* Run `migrate` and `collectstatic`
* Create a new import. This should add a Processing Layers block on the map, which can be expanded with a click. The spinner should be going for uploading and processing. Try closing the page. You should see an alert, but don't actually close it.
* Using `/admin/` change `status` to `processing` and save. This should change the spinner to a check for uploading.
* Using `/admin/` change `status` to `complete` and update the timestamp for `status_updated_at` and save. This should change the spinner to a check for processing.
* Refresh the page and the processing block should not be there.

Related Issue: #137

Connects #104
